### PR TITLE
An option to sort sections

### PR DESCRIPTION
**Pull request by funkyfuture**
this is a solution for the proposal in #90.

<del>from a user perspective, i'd actually rather prefer to simply provide a `True` to achieve simple alphabetical sorting. shall i extend to this?</del>

<del>should i add an example where a dict with weights to define the order is used?</del>

i also did some occasional modernization of code as one may consider it. are these changes okay?

### CODE REVIEW

**Comment by Chilipp** [comment]
Awesome! Thanks :smiley: I'll have a closer look tomorrow. But yes, I agree that a simple `True` option would be a good idea :+1:

File: docs/conf_settings.rst [hunk 1]
```diff
@@ -135,3 +135,23 @@ Configuration values and events
...
+
+   When ``True`` the summarizing sections will be sorted alphanumerically by
+   their section title. Alternatively a callable can be set that is passed to
+   :func:`sorted` as key argument to sort the the summary sections by their
+   name.
+   The default value is ``None``.
+   Example usage with a tuple that defines an arbitrary order:
+
+       sections_order = ("Public methods", "Attributes", "Private methods")
+       autodocsumm_section_sorter = lambda title: sections_order.index(title)
```
Comment (Chilipp) [inline docs/conf_settings.rst:150]:
```suggestion
   .. code-block:: python

       sections_order = ("Public methods", "Attributes", "Private methods")
       autodocsumm_section_sorter = lambda title: sections_order.index(title)
```

File: docs/conf_settings.rst [hunk 2]
```diff
@@ -135,3 +135,23 @@ Configuration values and events
...
+
+   An example for cases that only ensures that "Constructors" are always listed
+   first and "Attributes" while not failing when encountering undefined section
+   weights:
+
+       section_weights = {"Attributes": 100, "Constructors": -100}
+       autodocsumm_section_sorter = lambda title: sections_weights.get(title, 0)
```
Comment (Chilipp) [inline docs/conf_settings.rst:157]:
```suggestion
   .. code-block:: python

       section_weights = {"Attributes": 100, "Constructors": -100}
       autodocsumm_section_sorter = lambda title: sections_weights.get(title, 0)
```

**Review by Chilipp — CHANGES_REQUESTED** [review]
hey @funkyfuture! thanks a lot for the input here! The changes look great! I only have two minor suggestions for the docs, but otherwise I am happy to merge this when you are done here and make a new release :+1:

**Comment by Chilipp** [comment]
please let me know @funkyfuture if there is anything else from your side or if I can merge this PR as it is

**Comment by funkyfuture** [comment]
thanks for the feedback. i'm happy with the solution as it is.

**Review by Chilipp — APPROVED** [review]
awesome! :partying_face:

**Pull request merged**